app/src/main/java/com/android/sample/ui/utils/ValueControl.kt

```diff
@@ -0,0 +1,178 @@
+package com.android.sample.ui.utils
+
+import android.annotation.SuppressLint
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ExperimentalLayoutApi
+import androidx.compose.foundation.layout.FlowRow
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Checkbox
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.RangeSlider
+import androidx.compose.material3.SliderDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateMapOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import com.android.sample.model.filter.FloatRange
+import com.android.sample.resources.C.Dimension.PADDING_16
+import com.android.sample.resources.C.Dimension.PADDING_4
+import com.android.sample.resources.C.Dimension.PADDING_8
+import com.android.sample.resources.C.Tag.FilterPage.MAX_ITEM_IN_ROW
+import com.android.sample.resources.C.Tag.FilterPage.SLIDER_COLOR_ACTIVE
+import com.android.sample.resources.C.Tag.FilterPage.SLIDER_COLOR_INACTIVE
+import kotlinx.coroutines.flow.StateFlow
+
```

Comment:
- Splitting these utils into `ValueControl.kt` is a great idea—keeps the code modular and clean. 👍  
- `ValueRangeSlider` and `CheckboxGroup` are well-documented and reusable. No issues there. 👌  


Looks solid overall! 😊

Reply:
Thanks